### PR TITLE
[ML] MSE loss function for incremental training

### PR DESCRIPTION
**Pull request by tveasey**
This adds a new adjusted MSE loss function to penalise changing predictions from the tree being retrained.

### CODE REVIEW

**Comment by tveasey** [comment]
This still needs unit testing, but the code changes are ready to review.

**Review by valeriy42 — APPROVED** [review]
LGTM

File: include/maths/CBoostedTreeLoss.h [hunk 1]
```diff
@@ -55,18 +59,54 @@ class MATHS_EXPORT CArgMinLossImpl {
...
+private:
+    TMeanAccumulator m_MeanError;
+};
+
+//! \brief Finds the value to add to a set of predictions which minimises the
+//! adjusted regularized MSE difference w.r.t. the actual values for incremental
+//! training.
+//!
+//! DESCRIPTION:\n
+//! This applies a correction to the loss based on the difference between from
```
[the remark this answers is not in the document]
Comment (tveasey) [reply include/maths/CBoostedTreeLoss.h]: [e08b1e7](https://github.com/elastic/ml-cpp/pull/1793/commits/e08b1e700fdc94762494d3c31ab30618c34bba81)

**Comment by tveasey** [comment]
Failed with the known windows CI worker issue.

**Pull request merged**